src/main/kotlin/com/cultureamp/kotlin_kafka_http_sink_connector/main.kt

```diff
@@ -1,5 +1,149 @@
+    }
+
+    override fun taskClass(): Class<out Task> {
+        return HttpSinkTask::class.java
+    }
+
+    override fun taskConfigs(maxTasks: Int): MutableList<MutableMap<String, String>> {
+        val config: MutableMap<String, String> = mutableMapOf()
+        val topic = this.topic
+        val endpoint = this.endpoint
+        if (topic != null) {
+            config[TOPICS_CONFIG] = topic;
+        }
+        if (endpoint != null) {
+            config[CONFIG_ENDPOINT] = endpoint;
+        }
+        config[CONFIG_MAX_BATCH_SIZE] = "$maxBatchSize";
+        return MutableList(maxTasks) { config }
+    }
+
+    override fun version(): String {
+        return CONNECTOR_VERSION;
+    }
+}
+
+class HttpSinkTask : SinkTask {
+    private var topic: String? = null
+    private var endpoint: String? = null
+    private var maxBatchSize: Int = DEFAULT_MAX_BATCH_SIZE
+
+    constructor() : super()
+
+    override fun start(props: MutableMap<String, String>?) {
+        this.topic = props?.get(CONFIG_TOPIC)
+        this.endpoint = props?.get(CONFIG_ENDPOINT)
+        if (props?.contains(CONFIG_MAX_BATCH_SIZE) == true) {
+            this.maxBatchSize = Integer.parseInt(props[CONFIG_MAX_BATCH_SIZE]);
+        }
+    }
+
+    override fun stop() {
+        // Unless we start batching requests there's probably nothing we have to do here.
+    }
+
+    override fun put(records: MutableCollection<SinkRecord>?) {
+        if (records == null) {
+            return
+        }
+        println("Put records ${records.size}")
+        println("Config $topic $endpoint $maxBatchSize")
+        var dataToSend: String = ""
+        for (record in records) {
+            record.topic()
+            record.kafkaPartition()
+            record.kafkaOffset()
+            record.timestamp()
+            record.key()
+            record.keySchema()
+            record.value()
+            record.valueSchema()
+            println("----")
+            println("key ${record.key()}")
+            println("topic ${record.topic()}")
+            println("value ${record.value()}")
+            dataToSend = "${dataToSend}\n\nTopic ${record.topic()} Key ${record.key()} Value ${record.value()}"
+        }
+        if (dataToSend.isNotEmpty()) {
+            makeHttpRequest(this.endpoint, dataToSend)
+        }
+    }
+
+    private fun makeHttpRequest(endpoint: String?, dataToSend: String) {
+        if (endpoint == null) {
+            return
+        }
+        // Despite the recommendation being that connector tasks generally run asynchronously,
+        // if we want to guarantee partition ordering, we probably need to wait here.
+        // At least that's what the confluent HTTP and Lambda sink connectors seem to do in "sync" mode.
+        // Open question: how do we handle only blocking one partition if there's an error? Or does Kafka Connect do that for us?
```

Comment:
Notes from our zoom chat:

Docs dump:
- https://docs.confluent.io/current/connect/devguide.html#connectors-and-tasks
- https://docs.confluent.io/current/connect/concepts.html#distributed-workers
- https://www.confluent.io/blog/kafka-connect-deep-dive-error-handling-dead-letter-queues/ (talks about error tolerance)
- https://www.confluent.io/blog/create-dynamic-kafka-connect-source-connectors/ (talks about taskConfig)

Experiments/branches to try out:
- Can we change how the partitions are assigned to tasks? Can we do a 1-for-1 assignment of partition to task?
- Are the existing error tolerance modes suitable for blocking messages for a particular partition, or are they all-or-nothing.